presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -189,10 +189,10 @@ private boolean shouldMergeWithMemory(long memorySize)
     @Override
     public void close()
     {
-        if (hashAggregationBuilder != null) {
-            hashAggregationBuilder.close();
-        }
         try (Closer closer = Closer.create()) {
+            if (hashAggregationBuilder != null) {
```

Comment:
Can `hashAggregationBuilder` even be null? When a new instance of `SpillableHashAggregationBuilder` is constructed `hashAggregationBuilder` is guaranteed to be non-null (through `rebuildHashAggregationBuilder()`), no?

Also when we rebuild a new hash agg builder (in `spillToDisk()`), I guess we should close the existing one so that it releases whatever it has acquired.